### ISSUE TITLE
Gravity Update Connection refused

### ISSUE THREAD

**Issue by Taks808** [pi-hole/docker-pi-hole#1314]
This is a: Possible bug or configuration issue   

## Details

Went to check on PiHole via the web interface for updates. Noticed there was an issue in the diagnostic tab. All of my block lists were not accessible for update. I copied the weblink and pasted into multiple web browsers and I am able to get to the lists. 

## Related Issues
- [ ] I have searched this repository/Pi-hole forums for existing issues and pull requests that look similar 

https://www.reddit.com/r/pihole/comments/jqf1g4/unable_to_download_gravity_connection_refused/
https://github.com/pi-hole/pi-hole/issues/3061
https://discourse.pi-hole.net/t/update-gravity-connection-refused/18848

## How to reproduce the issue 

1. Environment data
  * Operating System: 
 [i] Pi-hole Docker Container: 2023.01.10
[✓] Distro:  Debian
[✓] Version: 11
[✓] dig return code: 0
  * Hardware: Synology DSM 6.2
  * Kernel Architecture: 
  * Docker Install Info and version: 
    - Software source: Downloaded latest from Docker Registry
    - Supplimentary Software: NONE
  * Hardware architecture: x86_64

2. docker-compose.yml contents, docker run shell command, or paste a screenshot of any UI based configuration of containers here
{
   "cap_add" : null,
   "cap_drop" : null,
   "cmd" : "",
   "cpu_priority" : 50,
   "devices" : null,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : true,
   "enabled" : true,
   "entrypoint_default" : "/s6-init",
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/opt/pihole:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "phpver",
         "value" : "php"
      },
      {
         "key" : "PHP_ERROR_LOG",
         "value" : "/var/log/lighttpd/error-pihole.log"
      },
      {
         "key" : "IPv6",
         "value" : "False"
      },
      {
         "key" : "S6_KEEP_ENV",
         "value" : "1"
      },
      {
         "key" : "S6_BEHAVIOUR_IF_STAGE2_FAILS",
         "value" : "2"
      },
      {
         "key" : "S6_CMD_WAIT_FOR_SERVICES_MAXTIME",
         "value" : "0"
      },
      {
         "key" : "FTLCONF_LOCAL_IPV4",
         "value" : "192.168.5.100"
      },
      {
         "key" : "FTL_CMD",
         "value" : "no-daemon"
      },
      {
         "key" : "DNSMASQ_USER",
         "value" : "root"
      },
      {
         "key" : "DNSMASQ_LISTENING",
         "value" : "local"
      },
      {
         "key" : "WEBPORT",
         "value" : "8080"
      },
      {
         "key" : "WEBPASSWORD",
         "value" : "sanitized"
      },
      {
         "key" : "TZ",
         "value" : "Pacific/Honolulu"
      }
   ],
   "exporting" : false,
   "id" : "e8e0f6a060f2709626183ca969da64dd1280dfdfe7e7cfef757a0d9f111ac282",
   "image" : "pihole/pihole:latest",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "pihole",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "bridge",
   "port_bindings" : [
      {
         "container_port" : 53,
         "host_port" : 53,
         "type" : "tcp"
      },
      {
         "container_port" : 53,
         "host_port" : 53,
         "type" : "udp"
      },
      {
         "container_port" : 80,
         "host_port" : 8080,
         "type" : "tcp"
      }
   ],
   "privileged" : true,
   "shortcut" : {
      "enable_shortcut" : false
   },
   "use_host_network" : false,
   "volume_bindings" : [
      {
         "host_volume_file" : "/docker/pihole2/pihole",
         "mount_point" : "/etc/pihole",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/pihole2/dnsmasq.d",
         "mount_point" : "/etc/dnsmasq.d",
         "type" : "rw"
      }
   ]
}

4. any additional info to help reproduce
Sorry no....

## These common fixes didn't work for my issue

- [X ] I have tried removing/destroying my container, and re-creating a new container
- [X] I have tried fresh volume data by backing up and moving/removing the old volume data
- [X] I have tried running the stock `docker run` example(s) in the readme (removing any customizations I added)
- [X] I have tried a newer or older version of Docker Pi-hole (depending what version the issue started in for me)
- [ ] I have tried running without my volume data mounts to eliminate volumes as the cause

If the above debugging / fixes revealed any new information note it here.
Add any other debugging steps you've taken or theories on root cause that may help.

I tried to use the web interface to create eh debug log and send the key in this help/bug request but i get an error 
 Error message: curl: (28) Failed to connect to tricorder.pi-hole.net port 443: Connection timed out

[✗] There was an error uploading your debug log.
   * Please try again or contact the Pi-hole team for assistance.
   * A local copy of the debug log can be found at: /var/log/pihole/pihole_debug.log
[Pihole Debug.txt](https://github.com/pi-hole/docker-pi-hole/files/10680121/Pihole.Debug.txt)

**Comment by hex-developer** [comment]
I also experience the same issue with updating the database.

For a workaround:
- Try downloading the list with wget
- Install python in the container
- Run `python -m http.server` in the directory you downloaded the list to. It should be accesible at localhost:8000/listname from the container, where listname is the name of the file you downloaded.
- While that is running add that url as an adlist and run an update
- The domains will be saved, but the next updates will fail.

If you want to update the database, you have to do this again.

**Comment by darrenmurphy7** [comment]
This issue is intermittent and reoccurring. I have tried various suggestions including sudo systemctl restart unbound, clearing the ftl cache and restarting the system.  Sometimes the issue is cleared without any reason as to why and then it randomly reappears.  Is there any update to fix this please?

**Comment by darrenmurphy7** [comment]
This issue is intermittent and reoccurring. I have tried various suggestions including sudo systemctl restart unbound, clearing the ftl cache, restarting the system and others. Sometimes the issue is cleared without any reason as to why and then it randomly reappears. Is there any update to fix this please?

**Comment by hex-developer** [comment]
@darrenmurphy7 I think the cause of the issue might be DNS not working in the container. I think docker has an option to specify a DNS server for the container. Try using your router's default DNS for that.

**Comment by PromoFaux** [comment]
Yeah, I think that is the solution.

In compose set a `dns: 8.8.8.8` or whatever your preferred resolver is.

Long ago we used to recommend the container having 127.0.0.1 as it's resolver but that can cause issues if FTL is not running, so it's not advisable to use that!

**Comment by darrenmurphy7** [comment]
> @darrenmurphy7 I think the cause of the issue might be DNS not working in the container. I think docker has an option to specify a DNS server for the container. Try using your router's default DNS for that.

Thank you for replying.. FYI I did not install via docker, my router dns points to unbound on pihole and unbound uses 127.0.0.1.  I tried adding cloudflare dns to pihole without success but when I added cloudflare dns as additional to router, pihole has successfully updated, thank you.
I dont want to use any other dns except unbound. Will there be a permanent fix for pihole gravity update in the future?
